xbmc/addons/AddonRepos.cpp

```diff
@@ -180,16 +180,37 @@ void CAddonRepos::AddAddonIfLatest(
 
 void CAddonRepos::BuildUpdateList(const std::vector<std::shared_ptr<IAddon>>& installed,
                                   std::vector<std::shared_ptr<IAddon>>& updates) const
+{
+  BuildUpdateOrOutdatedList(installed, updates, false);
+}
+
+void CAddonRepos::BuildOutdatedList(const std::vector<std::shared_ptr<IAddon>>& installed,
+                                    std::vector<std::shared_ptr<IAddon>>& outdated) const
+{
+  BuildUpdateOrOutdatedList(installed, outdated, true);
+}
+
+void CAddonRepos::BuildUpdateOrOutdatedList(const std::vector<std::shared_ptr<IAddon>>& installed,
+                                            std::vector<std::shared_ptr<IAddon>>& result,
+                                            bool returnOutdatedAddons) const
 {
   std::shared_ptr<IAddon> update;
 
-  CLog::Log(LOGDEBUG, "ADDONS: *** building update list (installed add-ons) ***");
+  CLog::Log(LOGDEBUG, "CAddonRepos::{}: Building {} list from installed add-ons", __func__,
+            returnOutdatedAddons ? "outdated" : "update");
 
   for (const auto& addon : installed)
   {
     if (DoAddonUpdateCheck(addon, update))
     {
-      updates.emplace_back(update);
+      if (!returnOutdatedAddons)
+      {
+        result.emplace_back(update);
+      }
+      else
+      {
+        result.emplace_back(addon);
+      }
```

Comment:
Could be simplified: => `result.emplace_back(returnOutdatedAddons ? addon : update);`